### ISSUE TITLE
Address domain bucket conflict with first field offset

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#998]
This is an issue which wouldn't be fixed by #967 and occurs in brubeck of https://github.com/goblint/bench/pull/53.

An `Lattice.Uncomparable` exception is raised from the joining of offsets (last case):
https://github.com/goblint/analyzer/blob/7f7b9f041c7ad51a4d7c3b51e5c19f2687ef9287/src/cdomains/lval.ml#L127-L138
The comment claims that the case is no longer used with buckets, but turns out it may be.

In particular, consider the join of bucketed address sets
```
((alloc@sid:628@tid:[main](#top)), concrat/brubeck/main.c:1841:3-1841:12).sampler ->
	((alloc@sid:628@tid:[main](#top)), concrat/brubeck/main.c:1841:3-1841:12).sampler
```
and
```
((alloc@sid:628@tid:[main](#top)), concrat/brubeck/main.c:1841:3-1841:12) ->
	((alloc@sid:628@tid:[main](#top)), concrat/brubeck/main.c:1841:3-1841:12).backend
```
The offset join is called on `.sampler` and `.backend` when the buckets are being merged.

This is surprising because the keys (representatives) of the buckets appear different: one has no offset and the other has `.sampler`.
Although the address sets buckets intend for them to be separate, the equality of offsets still uses the old special-cased logic:
https://github.com/goblint/analyzer/blob/7f7b9f041c7ad51a4d7c3b51e5c19f2687ef9287/src/cdomains/lval.ml#L92-L105
Thus, when the maps are being joined, the keys are found to be equal and hence their values are attempted to be joined.

Therefore, to fix the issue, the special cased logic should be stripped out completely (which #967 does not yet do!).
